### ISSUE TITLE
Mechanize::page#encoding_error? raises "ArgumentError: invalid byte sequence in UTF-8" when parse error message contains wrong characters.

### ISSUE THREAD

**Issue by nejiko96** [sparklemotion/mechanize#553]
I think this will be avoided by a patch like this:

```
--- a/lib/mechanize/page.rb
+++ b/lib/mechanize/page.rb
@@ -103,7 +103,7 @@ class Mechanize::Page < Mechanize::File
     parser = self.parser unless parser
     return false if parser.errors.empty?
     parser.errors.any? do |error|
-      error.message =~ /(indicate\ encoding)|
+      error.message.scrub =~ /(indicate\ encoding)|
                         (Invalid\ char)|
                         (input\ conversion\ failed)/x
     end
```

but it seems like String#scrub requires Ruby version >= 2.1.0.

**Comment by flavorjones** [comment]
@nejiko96 Can you help me reproduce the error you'd like to fix?

**Comment by nejiko96** [comment]
```
      agent = Mechanize.new
      page = agent.get('https://atcoder.jp/contests/apg4b/tasks/APG4b_r')
      page.search('body')
```

**Comment by flavorjones** [comment]
Thank you for your help. Working on a fix now.

**Comment by flavorjones** [comment]
I've released v2.8.1 which fixes this. Thanks again.

**Comment by nejiko96** [comment]
I have confirmed that it is fixed. Thank you very much.
